qt/scientific_interfaces/Indirect/IndirectSettingsPresenter.cpp

```diff
@@ -5,53 +5,35 @@
 //   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
 // SPDX - License - Identifier: GPL - 3.0 +
 #include "IndirectSettingsPresenter.h"
+#include "IndirectSettings.h"
 #include "IndirectSettingsHelper.h"
 
 namespace MantidQt::CustomInterfaces {
 using namespace IndirectSettingsHelper;
 
-IndirectSettingsPresenter::IndirectSettingsPresenter(QWidget *parent)
-    : QObject(nullptr), m_model(std::make_unique<IndirectSettingsModel>()),
-      m_view(std::make_unique<IndirectSettingsView>(parent)) {
-  setUpPresenter();
-}
-
-IndirectSettingsPresenter::IndirectSettingsPresenter(IndirectSettingsModel *model, IIndirectSettingsView *view)
-    : QObject(nullptr), m_model(model), m_view(view) {
-  setUpPresenter();
-}
-
-void IndirectSettingsPresenter::setUpPresenter() {
-  connect(m_view.get(), SIGNAL(okClicked()), this, SLOT(applyAndCloseSettings()));
-  connect(m_view.get(), SIGNAL(applyClicked()), this, SLOT(applyChanges()));
-  connect(m_view.get(), SIGNAL(cancelClicked()), this, SLOT(closeDialog()));
-
+IndirectSettingsPresenter::IndirectSettingsPresenter(std::unique_ptr<IndirectSettingsModel> model,
+                                                     IIndirectSettingsView *view)
+    : m_model(std::move(model)), m_view(view) {
+  m_view->subscribePresenter(this);
   loadSettings();
-
-  // Temporary until better validation is used when loading data into interfaces
-  setDefaultRestrictData();
 }
 
-void IndirectSettingsPresenter::setDefaultRestrictData() const {
-  auto const isisFacility = m_model->getFacility() == "ISIS";
-  if (isisFacility)
-    setRestrictInputDataByName(isisFacility);
-}
+QWidget *IndirectSettingsPresenter::getView() { return m_view->getView(); }
 
-IIndirectSettingsView *IndirectSettingsPresenter::getView() { return m_view.get(); }
+void IndirectSettingsPresenter::subscribeParent(IIndirectSettings *parent) { m_parent = parent; }
```

Comment:
I found this method a bit confusing. From what I gathered, here m_parent is a private attribute of the class, and not a fancy way of enforcing inheritance, right? So here any methods from the m_parent class need to be called as m_parent->method, and are not available from inheritance, correct?

Reply:
Yes you are right, `m_parent` does not refer to anything which is inherited. In this context, `m_parent` refers to the owner of the `IndirectSettingsPresenter` (i.e. the class that creates and holds a unique pointer to the `IndirectSettingsPresenter`)